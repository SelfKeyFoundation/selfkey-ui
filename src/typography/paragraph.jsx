import * as React from 'react';
import { css, withStyles } from '../default/with-styles';


class P extends React.Component {
    constructor(props) {
      super(props);
    }
  
    render() {
        const {
            styles,
            } = this.props;
        return (
            <p {...css(styles.Paragraph)}>
                BLOCKCHAIN IDENTITY WALLET
            </p>
        );
    }
  }

export default withStyles(({ selfkeyWallet: { color } }) => ({
    Paragraph: {
        color: '#93b0c1',
        fontFamily: '"Proxima Nova"',
        fontSize: '18px',
        fontWeight: 400,
        lineHeight: 1.67
    },


}))(P);

