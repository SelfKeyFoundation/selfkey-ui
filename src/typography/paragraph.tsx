import * as React from 'react';
// @ts-ignore
import injectSheet, { StyleSheet, StyledComponentProps } from 'react-jss';

const styles: StyleSheet = {
    p: {
        color: '#fff',
        fontFamily: 'Lato, arial, sans-serif',
        fontSize: '16px',
        fontWeight: 400,
        lineHeight: 1.67
    }
};

export type ParagraphProps = {
    className?: string,
}

export const P = injectSheet(styles)<ParagraphProps>(({ classes, children, className }) => (
    <p className={`${classes.base} ${classes.p} ${className}`}>{children}</p>
));
