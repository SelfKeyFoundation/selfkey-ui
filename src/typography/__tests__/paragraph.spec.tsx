import * as React from 'react';
import { create } from 'react-test-renderer';
import { P } from '../paragraph';


describe('Paragraph', () => {
  it('should show Paragraph', () => {
    const component = create(
      <P>TEST</P>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});





