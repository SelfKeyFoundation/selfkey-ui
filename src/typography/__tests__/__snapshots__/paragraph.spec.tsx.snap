// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should show Paragraph 1`] = `
<p
  className="undefined Component-p-0-1-1 undefined"
>
  TEST
</p>
`;
